crates/starknet-types-core/src/chain_id/alloc_impls.rs

```diff
@@ -0,0 +1,191 @@
+#[cfg(not(feature = "std"))]
+pub extern crate alloc;
+#[cfg(not(feature = "std"))]
+use alloc::string::{String, ToString};
+
+use crate::short_string;
+use crate::short_string::ShortString;
+
+use super::{ChainId, SN_MAIN_STR, SN_SEPOLIA_STR};
+
+impl From<ChainId> for ShortString {
+    fn from(value: ChainId) -> Self {
+        match value {
+            ChainId::Mainnet => short_string!("SN_MAIN"),
+            ChainId::Sepolia => short_string!("SN_SEPOLIA"),
+            #[cfg(feature = "devnet")]
+            ChainId::Devnet(ss) => ss,
+        }
+    }
+}
+
+#[cfg(feature = "devnet")]
+impl From<ShortString> for ChainId {
+    fn from(value: ShortString) -> Self {
+        if value.as_ref() == SN_MAIN_STR {
+            ChainId::Mainnet
+        } else if value.as_ref() == SN_SEPOLIA_STR {
+            ChainId::Sepolia
+        } else {
+            ChainId::Devnet(value)
+        }
+    }
+}
+
+#[cfg(not(feature = "devnet"))]
+mod try_chain_id_from_short_string {
+    use crate::chain_id::{SN_MAIN_STR, SN_SEPOLIA_STR};
+
+    use super::*;
+
+    #[derive(Debug, Clone)]
+    pub struct TryChainIdFromShortStringError(ShortString);
+
+    impl core::fmt::Display for TryChainIdFromShortStringError {
+        fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+            write!(f, "unknown chain id: {}", self.0)
+        }
+    }
+
+    #[cfg(feature = "std")]
+    impl std::error::Error for TryChainIdFromShortStringError {}
+
+    impl TryFrom<ShortString> for ChainId {
+        type Error = TryChainIdFromShortStringError;
+
+        fn try_from(value: ShortString) -> Result<Self, Self::Error> {
+            if value.as_ref() == SN_MAIN_STR {
+                Ok(ChainId::Mainnet)
+            } else if value.as_ref() == SN_SEPOLIA_STR {
+                Ok(ChainId::Sepolia)
+            } else {
+                Err(TryChainIdFromShortStringError(value))
+            }
+        }
+    }
+}
+#[cfg(not(feature = "devnet"))]
+pub use try_chain_id_from_short_string::*;
+
+// String
+
+impl From<ChainId> for String {
+    fn from(value: ChainId) -> Self {
+        match value {
+            ChainId::Mainnet => SN_MAIN_STR.to_string(),
+            ChainId::Sepolia => SN_SEPOLIA_STR.to_string(),
+            #[cfg(feature = "devnet")]
+            ChainId::Devnet(ss) => ss.to_string(),
+        }
+    }
+}
+
+#[cfg(not(feature = "devnet"))]
+impl From<ChainId> for &str {
+    fn from(value: ChainId) -> Self {
+        match value {
+            ChainId::Mainnet => SN_MAIN_STR,
+            ChainId::Sepolia => SN_SEPOLIA_STR,
+        }
+    }
+}
```

Comment:
Why conversion from `ChainId` to string is allowed only if `devnet` feature is disabled?